php/admin/class-migration.php

```diff
@@ -0,0 +1,210 @@
+							esc_html__( 'Genesis Custom Blocks', 'block-lab' )
+						)
+					);
+					?>
+				</p>
+				<p>
+					<?php
+					printf(
+						'<a target="_blank" href="https://getblocklab.com/docs/genesis-custom-blocks">%1$s</a>',
+						esc_html__( 'Learn more', 'block-lab' )
+					);
+					?>
+				</p>
+			</div>
+			<a href="#" id="bl-notice-not-now" class="bl-notice-option">
+				<?php esc_html_e( 'Not Now', 'block-lab' ); ?>
+			</a>
+			<a href="<?php echo esc_url( $migration_url ); ?>" class="bl-notice-option">
+				<?php esc_html_e( 'Migrate', 'block-lab' ); ?>
+			</a>
+		</div>
+		<div id="bl-not-now-notice" class="notice notice-info bl-notice-migration bl-hidden">
+			<div class="bl-migration-copy">
+				<p><?php esc_html_e( "When you're ready, our migration tool is available in the main menu, under Block Lab > Migrate.", 'block-lab' ); ?></p>
+			</div>
+			<a href="#" id="bl-notice-ok" class="bl-notice-option">
+				<?php esc_html_e( 'Okay', 'block-lab' ); ?>
+			</a>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Enqueues the migration notice assets.
+	 */
+	public function enqueue_assets() {
+		if ( ! $this->should_display_migration_notice() ) {
+			return;
+		}
+
+		wp_enqueue_style(
+			self::NOTICE_STYLE_SLUG,
+			$this->plugin->get_url( 'css/admin.migration-notice.css' ),
+			[],
+			$this->plugin->get_version()
+		);
+
+		wp_enqueue_script(
+			self::NOTICE_SCRIPT_SLUG,
+			$this->plugin->get_url( 'js/admin.migration-notice.js' ),
+			[],
+			$this->plugin->get_version(),
+			true
+		);
+	}
+
+	/**
+	 * Gets whether the migration notice should display.
+	 *
+	 * This should display on Block Lab > Content Blocks,
+	 * /wp-admin/plugins.php, the Dashboard, and Block Lab > Settings.
+	 *
+	 * @return bool Whether the migration notice should display.
+	 */
+	public function should_display_migration_notice() {
+		if ( ! current_user_can( self::NOTICE_CAPABILITY ) ) {
+			return false;
+		}
+
+		// If the user has dismissed the notice, it shouldn't appear again.
+		if ( self::NOTICE_DISMISSED_META_VALUE === get_user_meta( get_current_user_id(), self::NOTICE_USER_META_KEY, true ) ) {
+			return false;
+		}
+
+		$screen = get_current_screen();
+		return (
+			( isset( $screen->base, $screen->post_type ) && 'edit' === $screen->base && 'block_lab' === $screen->post_type )
+			||
+			( isset( $screen->base ) && in_array( $screen->base, [ 'plugins', 'dashboard', 'block_lab_page_block-lab-settings' ], true ) )
+		);
+	}
+
+	/**
+	 * Handles an AJAX request to not display the notice.
+	 *
+	 * This stores in the user meta the fact that the notice was dismissed,
+	 * so it's not displayed again.
+	 */
+	public function ajax_handler_migration_notice() {
+		check_ajax_referer( self::NOTICE_NONCE_ACTION, self::NOTICE_NONCE_NAME );
+
+		if ( ! current_user_can( self::NOTICE_CAPABILITY ) ) {
+			wp_send_json_error();
+		}
+
+		update_user_meta( get_current_user_id(), self::NOTICE_USER_META_KEY, self::NOTICE_DISMISSED_META_VALUE );
+
+		wp_die( 1 );
```

Comment:
This should be `wp_send_json_success()`, I think.

Reply:
Right, usually AJAX callbacks it would use that.

But this doesn't send anything back to the [JS logic that called it](https://github.com/getblocklab/block-lab/pull/635/files#diff-682871fb6fb8d69d6b84890ae345f685R16).

Reply:
I understand that we wouldn't be using any success response. It might be more efficient to just `wp_die()`.

Either way, we should try to be consistent. Use `wp_die()` for both the error and success cases (as it was prior to my `wp_send_json_error()` suggestion), or use `wp_send_json_success()` here. We don't have to provide any data, we can call that function without any parameters.

The only difference is that the unused output would be:
`{'success':true}`

Instead of:
`1`

Reply:
Sounds good, how about just `wp_die()` for both?

Reply:
Still, good point that it's still sending `1` or `-1` in the response.

Reply:
https://github.com/getblocklab/block-lab/pull/635/commits/4e8167a4e178d46c4bd26852b73e6663183f1667 changes this to `wp_send_json_success()`

Reply:
Thanks for 4e8167a – looks good. Would have been happy with `wp_die()`, but I like this approach better, personally, because it feels consistent with other AJAX responses.

Reply:
Thanks, it is more like other AJAX calls 😄